### ISSUE TITLE
When weight is not none, F.cross_entropy(reduction='none').mean() != F.cross_entropy()

### ISSUE THREAD

**Issue by mitming** [pytorch/pytorch#65329]
when `weight` is not None:
```
loss = F.cross_entropy(input, target, weight=self.per_cls_weights, reduction='none').mean()
``` 
is not equal to
```
loss = F.cross_entropy(input, target, weight=self.per_cls_weights)
```
or
```
loss = F.cross_entropy(input, target, weight=None)
```

I am really confused, how can I compute the final loss for each example?

![image](https://user-images.githubusercontent.com/73068772/134004513-f24d355f-2121-4ba5-8830-b736ddbf8214.png)

**Comment by ezyang** [comment]
I'm not going to close this immediately since I'm not sure if this is actually supported, but if you ask in https://discuss.pytorch.org/ someone is more likely to know if it is supported or not
